### PR TITLE
🍰 2019/kw17/release post2

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pullrequest

- [X] Beiträge können von deaktiviert zu aktiviert gesetzt werden.
- [X] Posts can be reenabled once disabled.

### Issues

- [X] fixes https://github.com/Human-Connection/Human-Connection/issues/432

### Checklist

- [X] None

### How2Test

- [X] None

### Todo

- [ ] backend implementation is missing

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "build": "babel src/ -d dist/ --copy-files",
     "start": "node dist/",
-    "dev": "nodemon --exec babel-node src/ -e js,graphql",
+    "dev": "nodemon --exec cross-env DISABLED_MIDDLEWARES=permissions babel-node src/ -e js,graphql",
```
Comment (appinteractive) [inline backend/package.json]: Please do not deactivate the permissions in dev!!!!!!!!

File: backend/package.json~ [hunk 2]
```diff
@@ -0,0 +1,97 @@
+{
```
Comment (appinteractive) [inline backend/package.json~]: Why that file? Issues with your editor?

File: backend/src/jwt/decode.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ export default async (driver, authorizationHeader) => {
   try {
     const decoded = await jwt.verify(token, process.env.JWT_SECRET)
     id = decoded.sub
-  } catch {
+  } catch (e) {
```
Comment (appinteractive) [inline backend/src/jwt/decode.js]: Just as a consistency hint: e stands often for event, try to use err as an error shorthand.

File: dat.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (appinteractive) [inline dat.json]: What is that?

File: webapp/pages/profile/_id/_slug.vue [hunk 5]
```diff
@@ -255,31 +256,36 @@
...
+
+ 
+                <ds-flex-item  class="ds-tab-nav-item ">
+                <ds-space margin="small">
                 <!-- TODO: find better solution for rendering errors -->
-                <!--<no-ssr>
+                <no-ssr>
                       <ds-number :label="$t('profile.shouted')">
                         <hc-count-to slot="count" :end-val="user.shoutedCount"/>
                       </ds-number>
```
Comment (appinteractive) [inline webapp/pages/profile/_id/_slug.vue]: Please format your files!

File: dat.json (hunk 4 above)
Comment (ogerly) [reply dat.json]: i put the hc project into a dat project for testing purposes .. dat.json is a file that has to be in the root folder. 
i took it out

File: backend/package.json~ (hunk 2 above)
Comment (roschaefer) [inline backend/package.json~]: Please remove this file. For the future, `git add .` and `git diff --cached` is your friend, prior to `git commit`.

File: webapp/components/Modal/DisableModal.vue [hunk 6]
```diff
@@ -72,8 +72,8 @@ export default {
         this.$toast.success(this.$t('disable.success'))
         this.isOpen = false
         setTimeout(() => {
-          this.$emit('close')
-        }, 1000)
+          this.$emit('close'), location.reload()
```
Comment (roschaefer) [inline webapp/components/Modal/DisableModal.vue]: Please avoid two command on one line. Make it obvious:

```suggestion
          this.$emit('close')
          location.reload()
```

File: webapp/components/Modal/ReleaseModal.spec.js [hunk 7]
```diff
@@ -0,0 +1,150 @@
+import { shallowMount, mount, createLocalVue } from '@vue/test-utils'
+import DisableModal from './ReleaseModal.vue'
```
Comment (roschaefer) [inline webapp/components/Modal/ReleaseModal.spec.js]:
```suggestion
import ReleaseModal from '.'
```

Comment (roschaefer) [reply webapp/components/Modal/ReleaseModal.spec.js]: By now we have a convention that we place the component file and the test in one folder and name that folder according to the componnent name. Example:

```
RelaseModal/
  index.vue
  spec.js
```

Then you can import the component as above :point_up:

File: webapp/locales/de.json [hunk 8]
```diff
@@ -239,5 +239,26 @@
   },
   "shoutButton": {
     "shouted": "empfohlen"
+  },
+ 
+  "release": {
+    "submit": "freigeben",
+    "cancel": "Abbrechen",
+    "success": "Erfolgreich freigeben",
```
Comment (roschaefer) [inline webapp/locales/de.json]:
```suggestion
    "success": "Erfolgreich freigegeben",
```

File: webapp/locales/en.json [hunk 9]
```diff
@@ -247,5 +248,26 @@
   },
   "shoutButton": {
     "shouted": "shouted"
+  },
+  
+  "release": {
+    "submit": "Release",
+    "cancel": "Cancel",
+    "success": "Release successfully",
```
Comment (roschaefer) [inline webapp/locales/en.json]:
```suggestion
    "success": "Released successfully",
```

File: webapp/pages/profile/_id/_slug.vue [hunk 10]
```diff
@@ -252,31 +252,32 @@
                     </no-ssr>
                   </ds-space>
                 </ds-flex-item>
-                <!--<ds-flex-item class="ds-tab-nav-item">
-                <ds-space margin="small">-->
-                <!-- TODO: find better solution for rendering errors -->
-                <!--
-                    <no-ssr>
+                 
+                <!-- <ds-flex-item  v-tooltip="{content: 'Deine Beiträge', placement: 'left', delay: { show: 500 }}" class="ds-tab-nav-item ">
```
Comment (roschaefer) [inline webapp/pages/profile/_id/_slug.vue]: No German hard-coded texts

File: webapp/components/Modal/ReleaseModal.spec.js (hunk 7 above)
Comment (ogerly) [reply webapp/components/Modal/ReleaseModal.spec.js]: by arrangement we have agreed on this form

RelaseModal/
  RelaseModal.vue
  RelaseModal.spec.js

Comment (ogerly) [reply webapp/components/Modal/ReleaseModal.spec.js]: fixed

**Comment by ogerly** [comment]
when man the test is longer then there there's not least one problem with the test. 
"describe('after timeout', () => { {
   beforeEach(jest.runAllTimers)" " 

runAllTimers will the wohl one value

png

**Comment by mattwr18** [comment]
I had this error at one point as well @ogerly... I am trying to remember what I did to get passed it

**Comment by mattwr18** [comment]
here https://github.com/Human-Connection/Human-Connection/pull/525/files#diff-2ff74df6d7f961553d2d665605da64eeR76

where you call `location.reload()`, I think you need to mock this in your tests, like 
```
location: {
  reload: jest.fn()
}
```
could you add that and see if it still fails in the same way?

**Comment by mattwr18** [comment]
helpful article @ogerly 
https://michaelnthiessen.com/force-re-render

**Comment by ogerly** [comment]
clear  version from this https://github.com/Human-Connection/Human-Connection/pull/736

**Pull request closed without merging**